searx/data/engines_languages.json

```diff
@@ -1,964 +1,8 @@
-    "de-CH",
-    "de-DE",
-    "de-LI",
-    "dv-MV",
-    "el-GR",
-    "en-AG",
-    "en-AI",
-    "en-AS",
-    "en-AU",
-    "en-BB",
-    "en-BM",
-    "en-BS",
-    "en-BW",
-    "en-BZ",
-    "en-CA",
-    "en-CC",
-    "en-CK",
-    "en-CX",
-    "en-CY",
-    "en-DM",
-    "en-FJ",
-    "en-FK",
-    "en-FM",
-    "en-GB",
-    "en-GD",
-    "en-GG",
-    "en-GH",
-    "en-GI",
-    "en-GM",
-    "en-GU",
-    "en-GY",
-    "en-IE",
-    "en-IN",
-    "en-JE",
-    "en-JM",
-    "en-KI",
-    "en-KN",
-    "en-KY",
-    "en-LC",
-    "en-LR",
-    "en-LS",
-    "en-MH",
-    "en-MM",
-    "en-MP",
-    "en-MS",
-    "en-MU",
-    "en-MW",
-    "en-NF",
-    "en-NG",
-    "en-NR",
-    "en-NU",
-    "en-NZ",
-    "en-PG",
-    "en-PH",
-    "en-PN",
-    "en-PW",
-    "en-SB",
-    "en-SC",
-    "en-SG",
-    "en-SH",
-    "en-SL",
-    "en-SS",
-    "en-SZ",
-    "en-TC",
-    "en-TK",
-    "en-TO",
-    "en-TT",
-    "en-TV",
-    "en-TZ",
-    "en-UG",
-    "en-US",
-    "en-VC",
-    "en-VG",
-    "en-VI",
-    "en-VU",
-    "en-WS",
-    "en-ZA",
-    "en-ZM",
-    "en-ZW",
-    "es-AR",
-    "es-BO",
-    "es-CL",
-    "es-CO",
-    "es-CR",
-    "es-DO",
-    "es-EC",
-    "es-ES",
-    "es-GQ",
-    "es-GT",
-    "es-HN",
-    "es-MX",
-    "es-NI",
-    "es-PA",
-    "es-PE",
-    "es-PR",
-    "es-PY",
-    "es-SV",
-    "es-US",
-    "es-UY",
-    "es-VE",
-    "et-EE",
-    "fa-IR",
-    "fi-FI",
-    "fo-FO",
-    "fr-BE",
-    "fr-BF",
-    "fr-BI",
-    "fr-BJ",
-    "fr-BL",
-    "fr-CA",
-    "fr-CD",
-    "fr-CF",
-    "fr-CG",
-    "fr-CH",
-    "fr-CI",
-    "fr-CM",
-    "fr-DJ",
-    "fr-FR",
-    "fr-GA",
-    "fr-GF",
-    "fr-GN",
-    "fr-GP",
-    "fr-HT",
-    "fr-KM",
-    "fr-LU",
-    "fr-MC",
-    "fr-MF",
-    "fr-MG",
-    "fr-ML",
-    "fr-MQ",
-    "fr-NC",
-    "fr-NE",
-    "fr-PF",
-    "fr-PM",
-    "fr-RE",
-    "fr-SN",
-    "fr-TD",
-    "fr-TG",
-    "fr-WF",
-    "fr-YT",
-    "he-IL",
-    "hr-HR",
-    "hu-HU",
-    "hy-AM",
-    "id-ID",
-    "is-IS",
-    "it-IT",
-    "it-SM",
-    "it-VA",
-    "ja-JP",
-    "ka-GE",
-    "kk-KZ",
-    "kl-GL",
-    "km-KH",
-    "ko-KR",
-    "ky-KG",
-    "lo-LA",
-    "lt-LT",
-    "lv-LV",
-    "mk-MK",
-    "mn-MN",
-    "ms-BN",
-    "ms-MY",
-    "mt-MT",
-    "nb-NO",
-    "ne-NP",
-    "nl-AW",
-    "nl-BE",
-    "nl-CW",
-    "nl-NL",
-    "nl-SR",
-    "nl-SX",
-    "pl-PL",
-    "prs-AF",
-    "pt-AO",
-    "pt-BR",
-    "pt-CV",
-    "pt-GW",
-    "pt-MZ",
-    "pt-PT",
-    "pt-ST",
-    "ro-MD",
-    "ro-RO",
-    "ru-RU",
-    "rw-RW",
-    "si-LK",
-    "sk-SK",
-    "sl-SI",
-    "so-SO",
-    "sq-AL",
-    "sr-LATN-RS",
-    "sr-ME",
-    "sv-SE",
-    "sw-KE",
-    "tg-CYRL-TJ",
-    "th-TH",
-    "ti-ER",
-    "tk-TM",
-    "tr-TR",
-    "uk-UA",
-    "ur-PK",
-    "uz-LATN-UZ",
-    "vi-VN",
-    "zh-CN",
-    "zh-HK",
-    "zh-MO",
-    "zh-TW"
-  ],
+  "bing": [],
+  "bing images": [],
+  "bing news": [],
+  "bing videos": [],
```

Comment:
Do not merge this PR / first I will have a look why bing languages do no longer work.

Reply:
The Request to, and the Response from https://www.bing.com/account/general has been changed ..

One issue I think is here in the Request ..

https://github.com/searxng/searxng/blob/6b49f32d3e66daa0bd0e7d37630a51968e0f6f64/searx/engines/__init__.py#L196


```diff
diff --git a/searx/engines/__init__.py b/searx/engines/__init__.py
index fa9749e9..3d104137 100644
--- a/searx/engines/__init__.py
+++ b/searx/engines/__init__.py
@@ -193,7 +193,7 @@ def set_language_attributes(engine):
     if hasattr(engine, '_fetch_supported_languages'):
         headers = {
             'User-Agent': gen_useragent(),
-            'Accept-Language': 'ja-JP,ja;q=0.8,en-US;q=0.5,en;q=0.3',  # bing needs a non-English language
+            'Accept-Language' : "en-US,en;q=0.5" # bing needs to set the English language
         }
         engine.fetch_supported_languages = (
             # pylint: disable=protected-access
```

The HTML response from www.bing.com/account/general has been changed, this function does no longer scrape out the languages:

https://github.com/searxng/searxng/blob/6b49f32d3e66daa0bd0e7d37630a51968e0f6f64/searx/engines/bing.py#L129-L144

Reply:
This issue is fixed in https://github.com/searxng/searxng/pull/676